elasticsearch.go

```diff
@@ -369,6 +397,22 @@ func (c *BaseClient) Perform(req *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 
+	if c.kerberosClient != nil {
+		if respUnauthorizedNegotiate(res) {
+			err := spnego.SetSPNEGOHeader(c.kerberosClient, req, "")
```

Comment:
I am not that familiar with kerberos auth protocol, could you help check my understanding of how this API is being used?

IIUC, the `SetSPNEGOHeader` would use the request to identify the principal name for the service but it should only require the metadata, so why do we need the tee reader here? Or is there another place where the request body will be consumed?